### PR TITLE
[WIP] Add Keen support for osf4m [#OSF-5979]

### PR DESCRIPTION
**Pull request by jeffreyliu3230**
## Purpose

Send osf4m logs to Keen server.

## Changes

New feature.

## Side effects

Changes add_poster_by_email function.

## Ticket

https://openscience.atlassian.net/browse/OSF-5979

### CODE REVIEW

File: website/conferences/views.py [hunk 1]
```diff
@@ -28,6 +29,22 @@
 
 logger = logging.getLogger(__name__)
 
+# Setup Keen client
+keen.project_id = settings.KEEN_PROJECT_ID
+keen.write_key = settings.KEEN_WRITE_KEY
+
+
+def _dt_to_iso8601(value):
```
Comment (icereval) [inline website/conferences/views.py]: @sloria, do we have a helper anywhere for this?

Comment (jeffreyliu3230) [inline website/conferences/views.py]: @icereval FYI i copied the function from here: https://github.com/CenterForOpenScience/osf.io/blob/develop/api_tests/files/views/test_file_detail.py#L21 and changed it a bit. I would guess there is no helper.

File: website/conferences/views.py [hunk 2]
```diff
@@ -138,6 +155,27 @@ def add_poster_by_email(conference, message):
...
+                "api": "/api/v1/profile/{}/".format(user._id)
+            },
+            "timezone": user.timezone,
+            "fullname": user.fullname,
+            "isActive": user.is_active,
+            "id": user._id
+        },
+        "meeting": message.conference_name,
+        "userCreated": user_created,
+        "timeSubmitted": _dt_to_iso8601(datetime.utcnow())
```
Comment (felliott) [inline website/conferences/views.py]: Can you make the above three attributes under an "osf4m" top-level key?  Could you also add a static field product.id?  `"product": { "id": "osf4m" }`

File: website/conferences/views.py [hunk 3]
```diff
@@ -138,6 +155,27 @@ def add_poster_by_email(conference, message):
     if node_created and user_created:
         signals.osf4m_user_created.send(user, conference=conference, node=node)
 
+    # Send log to Keen
+    keen.add_event('osf4m', {
+        "user": {
+            "username": user.username,
+            "userEmail": message.sender_email,
```
Comment (felliott) [inline website/conferences/views.py]: Please make this a subproperty of the "osf4m" key and rename it to "receivedFrom".

File: website/conferences/views.py (hunk 1 above)
Comment (icereval) [inline website/conferences/views.py]: Might be worth adding this to `osfHelpers.js` in the vicinity this other 8601 helper.

Comment (sloria) [inline website/conferences/views.py]: Would this work? https://github.com/CenterForOpenScience/osf.io/blob/8225f315d414e89dd72c0f277bca0f8a6823273e/framework/utils.py#L7-L9

Comment (felliott) [inline website/conferences/views.py]: @sloria, it would need millisecond support, but yes.  That function could use some timezone-verification, though.

**Comment by felliott** [comment]
The code is good, but @jeffreyliu3230 discovered that this information is already available through User `system_tags`, so this PR is no longer needed.

**Pull request closed without merging**